.github/workflows/scripts/builder-fetch.sh

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Caller sets the following:
```

Comment:
Note: We will replace this with Golang code in the future https://github.com/slsa-framework/slsa-github-generator/issues/87